apps/cowswap-frontend/src/modules/hooksStore/hooks/useBalancesDiff.ts

```diff
@@ -0,0 +1,124 @@
+import { useTenderlyBundleSimulation } from 'modules/tenderly/hooks/useTenderlyBundleSimulation'
+import { useOrderParams } from './useOrderParams'
+import { useHooks } from './useHooks'
+import { useMemo } from 'react'
+import { useWalletInfo } from '@cowprotocol/wallet'
+import { BalancesDiff } from 'modules/tenderly/types'
+import { BigNumber } from 'ethers'
+
+const EMPTY_BALANCE_DIFF: BalancesDiff = {}
+
+export function usePreHookBalanceDiff(): BalancesDiff {
+  const { data } = useTenderlyBundleSimulation()
+
+  const { preHooks } = useHooks()
+
+  return useMemo(() => {
+    if (!data || !preHooks.length) return EMPTY_BALANCE_DIFF
+
+    const lastPreHook = preHooks[preHooks.length - 1]
+    return data[lastPreHook?.uuid]?.cumulativeBalancesDiff || EMPTY_BALANCE_DIFF
+  }, [data, preHooks])
+}
+
+// Returns all the ERC20 Balance Diff of the current hook to be passed to the iframe context
+export function useHookBalancesDiff(isPreHook: boolean, hookToEditUid?: string): BalancesDiff {
+  const { account } = useWalletInfo()
+  const { data } = useTenderlyBundleSimulation()
+  const orderParams = useOrderParams()
+  const { preHooks, postHooks } = useHooks()
+  const preHookBalanceDiff = usePreHookBalanceDiff()
+
+  const orderMockBalanceDiff = useMemo(() => {
```

Comment:
why mock?

Reply:
Sorry for the lack of documentation. This is used mainly when the swap isn't simulated (like if the user has only preHooks). I changed a bit the variable name and comments.

Reply:
I see, thanks